tests/test_metadata.py

```diff
@@ -0,0 +1,284 @@
+        metadata_dict = yaml.safe_load(file)
+    context_dict = {
+        "eval": True,
+        "batch_size": 1,
+        "model_type": str(type(AutoConfig.from_pretrained("gpt2"))),
+        "use_token_type_ids": use_token_type_ids,
+        "vocab_size": 50257,
+        "number_of_workers": 4,
+        "max_seq_length": 1024,
+    }
+    try:
+        DatasetMetadata.model_validate(metadata_dict, context=context_dict)
+    except ValidationError as exc:
+        assert len(exc.errors()) == 2
+        for error in exc.errors():
+            assert error["loc"][0] in error_keys
+            if error["loc"][0] == "max_batch_size_dev":
+                assert (
+                    "Evaluation during training is turned on but there are no " "evaluation files in this dataset"
+                ) in error["msg"]
+            elif error["loc"][0] == "number_of_dev_files":
+                assert ("Evaluating during runtime but have no evaluation files to run in dataset") in error["msg"]
+            else:
+                assert False
+        return
+    assert False
+
+
+@pytest.mark.parametrize("use_token_type_ids", [(True), (False)])
+def test_pydantic_model_yes_evaluation_files(use_token_type_ids):
+    """Testing DatasetMetadata loads in variables correctly. This should pass"""
+    output_dir = os.path.join(
+        Path.cwd(),
+        "tests",
+        "examples",
+        "pretraining_split_with_new_metadata_and_sha256",
+    )
+    metadata_file = os.path.join(output_dir, "metadata.yaml")
+    with open(metadata_file, "r") as file:
+        metadata_dict = yaml.safe_load(file)
+    metadata_dict["number_of_dev_files"] = 1
+    metadata_dict["max_batch_size_dev"] = 7
+    context_dict = {
+        "eval": True,
+        "batch_size": 1,
+        "model_type": str(type(AutoConfig.from_pretrained("gpt2"))),
+        "use_token_type_ids": use_token_type_ids,
+        "vocab_size": 50257,
+        "number_of_workers": 4,
+        "max_seq_length": 1024,
+    }
+    DatasetMetadata.model_validate(metadata_dict, context=context_dict)
+
+
+def test_metadata_end2end_output(tmp_path):
+    """Testing the e2e pipeline is still the same. This should pass"""
+
+    tmp_e2e_output_dir = tmp_path / "e2e_testing_output_directory"
+    logging.info(f"temporary e2e output directory is in {tmp_e2e_output_dir}")
+    base_dir = os.path.join(Path.cwd(), "tests", "examples", "pretraining_split_with_new_metadata_and_sha256")
+    input_file = os.path.join(base_dir, "example_pretraining_data.jsonl")
+    command = [
+        "python3",
+        "-m",
+        "generative_data_prep",
+        "pipeline",
+        "--input_file_path",
+        f"{input_file}",
+        "--pretrained_tokenizer",
+        "gpt2",
+        "--max_seq_length",
+        "1024",
+        "--shuffle",
+        "on_RAM",
+        "--num_training_splits",
+        "4",
+    ]
+    subprocess.run(command + ["--keep_split_jsonls", "--output_path", f"{tmp_e2e_output_dir}"])
```

Comment:
This might be risky depending on if user has python3 defined, please update it to call pipeline_main directly

as in file below

https://github.com/sambanova/generative_data_prep/blob/main/tests/test_e2e.py


<img width="476" alt="image" src="https://github.com/sambanova/generative_data_prep/assets/137233191/92b71388-28bb-414a-925c-cf3b0f5d13c8">